src/OCIStorageService.js

```diff
@@ -0,0 +1,371 @@
+/**
+ * @file        - OCI Storage Service
+ * @exports     - `OCIStorageService`
+ * @since       - 5.0.1
+ * @version     - 1.0.0
+ * @implements  - BaseStorageService
+ *                OCI Storage support S3 compatible API Please see the S3 API implementation
+ * @see {@link https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html | X-Amz-Credential}
+ * @see {@link https://docs.aws.amazon.com/directconnect/latest/APIReference/CommonParameters.html#CommonParameters-X-Amz-Credential | X-Amz-Credential}
+ */
+
+const BaseStorageService  = require('./BaseStorageService');
+const { logger }          = require('@project-sunbird/logger');
+const _                   = require('lodash');
+const dateFormat          = require('dateformat');
+const uuidv1              = require('uuid/v1');
+const async               = require('async');
+const storageLogger       = require('./storageLogger');
+const { getSignedUrl }    = require("@aws-sdk/s3-request-presigner");
+const { S3Client, GetObjectCommand, HeadObjectCommand } = require("@aws-sdk/client-s3");
+const { Upload }          = require("@aws-sdk/lib-storage");
+const multiparty          = require('multiparty');
+export class OCIStorageService extends BaseStorageService {
```

Comment:
Lot of code of AWSStorageService & OCIStorageService(this file) looks same. If so why don't you extend the AWSStorageService by overriding the methods which requires custom changes for OCI. 
Or 
Create the Base case by adding common methods of AWS & OCI. Both the services can extend the same base class by overriding the methods which needs customisation.

Reply:
oci implementation is a direct copy of the aws implementation. So whatever existing logic is there in the aws implementation will be there in OCI. The only difference is that because oci is an s3 compliant storage, only the authentication part has to follow the s3 compliant specs. In this case it will be to provide endpoint , region, signing version etc.

Every thing else will be as is.

There will be no change in the method implementation.